lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -46,10 +62,59 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
+        def relevant_included_associations_for(serializer, nested_associations = {})
+          if nested_associations.present?
+            serializer.associations.select do |association|
+              # nested_associations is a hash of:
+              #   key => nested association to include
+              nested_associations.key?(association.name)
+            end
+          else
+            serializer.associations
+          end
+        end
+
+        # add a singular relationship
+        # the options should always belong to the serializer
+        def to_one_relationship_for(serializer, options)
+          if options[:include].present?
```

Comment:
That's not a good enough reason :) Why is it necessary, functionality-wise? What does it ensure?